### PR TITLE
Consolidate value matchers into single WitResponseMatcher

### PR DESCRIPTION
**Pull request by yolanother**

### CODE REVIEW

File: Scripts/Runtime/CallbackHandlers/WitResponseMatcher.cs [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+            if (!double.TryParse(value, out dValue)) return false;
+
+            // We will throw an exception if match value is not a numeric value. This is a developer
+            // error.
+            double dMatchValue = double.Parse(matchValue);
+
+            switch (comparisonMethod)
+            {
+                case ComparisonMethod.Equals:
+                    return Math.Abs(dValue - dMatchValue) < .0001f;
```
Comment (nick-benson) [inline Scripts/Runtime/CallbackHandlers/WitResponseMatcher.cs:141]: Const for this? Since it's used several times.

Comment (yolanother) [reply Scripts/Runtime/CallbackHandlers/WitResponseMatcher.cs:141]: Yeah I was debating adding a tolerance field. I think I will.

**Comment by yolanother** [comment]
image

**Comment by yolanother** [comment]
image
Cleaned up the Value Matchers field so it is more clear what fields are used when.

**Review by nick-benson — APPROVED** [review]

**Pull request merged**